src/main/java/com/google/sps/servlets/EditOrganizationServlet.java

```diff
@@ -18,10 +18,68 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.text.SimpleDateFormat;
 import com.google.sps.data.User;
+import com.google.sps.data.HistoryManager;
+import com.google.sps.data.OrganizationUpdater;
+import com.google.sps.data.GivrUser;
+import java.time.Instant;
 import java.io.IOException;
+import com.google.gson.Gson;
 
 @WebServlet("/edit-organization")
 public class EditOrganizationServlet extends HttpServlet {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    //Get the proper organization entity for updating
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    long organizationId = Long.parseLong(request.getParameter("id"));
+    Key organizationKey = KeyFactory.createKey("Distributor", organizationId);
+    Entity organizationEntity = new Entity("Distributor");
+
+    // try catch for compliation purposes, servlet will not be called without a valid id param
+    try {
+      organizationEntity = datastore.get(organizationKey);
+    } catch(com.google.appengine.api.datastore.EntityNotFoundException err) {
+        response.sendError(HttpServletResponse.SC_NOT_FOUND);
+        return;
+    }
+    
+    GivrUser user = GivrUser.getLoggedInUser();
```

Comment:
should you check if the user is logged in here and throw an error otherwise?

Reply:
Seems like a good function to add to GivrUser:

void ThrowIfNotLoggedIn() {
}

Reply:
Currently this check happens a bit earlier since the edit button will only be shown for logged in users but will put that function on the list of refactoring for GivrUser